code/modules/mob/living/carbon/alien/organs.dm

```diff
@@ -158,7 +158,37 @@
 	zone = BODY_ZONE_PRECISE_MOUTH
 	slot = ORGAN_SLOT_XENO_ACIDGLAND
 	actions_types = list(/datum/action/cooldown/alien/acid/corrosion)
+	/// How much acid this releases on the death of the holding mob
+	var/acid_release_on_death = 150
 
+/obj/item/organ/internal/alien/acid/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+
+	RegisterSignal(organ_owner, COMSIG_LIVING_DEATH, PROC_REF(release_acid))
+
+/obj/item/organ/internal/alien/acid/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+
+	UnregisterSignal(organ_owner, COMSIG_LIVING_DEATH)
+
+// Release acid on death and destroy all internal organs, except for the acid gland
+/obj/item/organ/internal/alien/acid/proc/release_acid()
+	if (!owner)
+		return
+
+	var/datum/reagents/releasing_acid = new(acid_release_on_death)
+	releasing_acid.add_reagent(/datum/reagent/toxin/acid, acid_release_on_death)
+	releasing_acid.my_atom = owner
+	releasing_acid.create_foam(/datum/effect_system/fluid_spread/foam, acid_release_on_death)
+
+	// Destroy all of the holders internal organs
+	for (var/obj/item/organ/internal/organ as anything in owner.organs)
+		if (organ == src)
+			continue
+		if (istype(organ, /obj/item/organ/internal/brain))
+			continue
+		organ.Remove(owner)
```

Comment:
Can organs have item flags which make them acidproof? Might be worth checking against those (and then applying them to this organ so you can skip the src check)

Reply:
Also TBH I think being able to remove xeno organs from most xenos is like... intentional.